test/e2e/operator_test.go

```diff
@@ -307,6 +307,32 @@ func TestExternalDNSRecordLifecycle(t *testing.T) {
 		t.Fatalf("Failed to get loadbalancer IPs for service %s/%s: %v", testNamespace, testServiceName, err)
 	}
 
+	// test for azure private zone
+	a, ok := helper.(*azureTestHelper)
+	if ok && a.isPrivateZone {
```

Comment:
I don't understand this section. So the DNS record for the loadbalancer in the Azure private zones will be published publicly? And then we have to veriify that they are also published on the Azure's own DNS servers?

Reply:
afaik, the difference here is in the resolver. Azure PrivateDNS does not return name servers, we use the `virtual network link` to connect directly to the private dns zone, and lookup the test server record. 

Please correct me if i am wrong

Reply:
IIUC the approach you are taking won't work because the link is being created between the test cluster and the private DNS zone. However the test does not run in the same test cluster. It could be running in another cluster which could be in another cloud environment.
Also I recommend you first stub out the test for Azure private DNS and enable the test in CI and then gradually enable the test. You will get faster feedback while implementing the tests.